languages/elixir/concepts/guards/about.md

```diff
@@ -1 +1,50 @@
-TODO: add information on guards concept
+[Guards][guards] are used as a complement to pattern matching. They allow for more complex checks. They can be used in [some, but not all situations][where-guards-can-be-used] where pattern matching can be used, for example in function clauses or case clauses.
+
+```elixir
+def empty?(list) when is_list(list) and length(list) == 0 do
+  true
+end
+```
+
+- Guards begin with the `when` keyword, followed by a boolean expression.
+- Guard expressions are special functions which:
+  - Must be [pure][pure-function] and not mutate any global states.
+  - Must return strict `true` or `false` values.
+- A list of common guards are found in the [Elixir documentation][kernel-guards]. They include:
+  - Type checks: [`is_integer/1`][guard-is-integer], [`is_list/1`][guard-is-list], [`is_nil/1`][guard-is-nil] etc.
+  - Arithmetic operators: [`+/2`][guard-plus], [`-/2`][guard-minus] etc.
+  - Comparisons: [`==/2`][guard-equals], [`>/2`][guard-greater], [`</2`][guard-less] etc.
+  - Binary operators: [`and/2`][guard-and], [`or/2`][guard-or], [`not/1`][guard-not]
+  - Membership operator: [`in/2`][guard-in]
+- You can define your own guard with [`defguard`][defguard].
+
+  - According to Elixir's [naming convention][naming], guard names should start with `is_`.
+
+    ```elixir
+    defmodule HTTP do
+      defguard is_success(code) when code >= 200 and code < 300
+
+      def handle_response(code) when is_success(code) do
+        :ok
+      end
+    end
+    ```
```

Comment:
I extended the original text about guards with an explanation where they can be used, examples of commonly used expression, and a mention of `defguard`.